qt/src/hocr/HOCRPdfExporter.cc

```diff
@@ -684,3 +712,43 @@ QImage HOCRPdfExporter::getSelection(const QRect& bbox)
 {
 	return m_displayerTool->getSelection(bbox);
 }
+
+void HOCRPdfExporter::paperSizeChanged() {
+	delete ui.lineEditPaperSizeHeight->validator();
+	delete ui.lineEditPaperSizeWidth->validator();
+
+	if(ui.comboBoxPaperSizeMetric->itemText(ui.comboBoxPaperSizeMetric->currentIndex()) == "mm") {
+		ui.lineEditPaperSizeHeight->setValidator(new QIntValidator(0, 10000000));
+		ui.lineEditPaperSizeWidth->setValidator(new QIntValidator(0, 10000000));
+	} else if(ui.comboBoxPaperSizeMetric->itemText(ui.comboBoxPaperSizeMetric->currentIndex()) == "inch") {
+		ui.lineEditPaperSizeHeight->setValidator(new QDoubleValidator(0.0, 10000000.0, 2));
+		ui.lineEditPaperSizeWidth->setValidator(new QDoubleValidator(0.0, 10000000.0, 2));
+	}
+
```

Comment:
You can just adjust the params of the validator instead of creating a new one each time.

Reply:
How can i do it, if in one case i must use QIntValidator, in another case - QDoubleValidator>

Reply:
Just `QDoubleValidator::setDecimals(0)`.

Reply:
But in this case i should get validator from QLineEdit. QLineEdit returns only const QValidator* , so i can't change validator by this pointer.

Reply:
Just static-cast it, you know what type of validator the line edit has.

Reply:
I found one problem: if i set to QDoubleValidator setDecimals() to 0, QDoubleValidator allows to user enter single comma/dot without any decimal part. QIntValidator disallows.

Reply:
I don't really see a problem with that.

Reply:
But QIntValidator is more convinient for "mm", and QDoubleValidator is more convinient for "inch" modes. Can you suggest more beautiful way deal with it? (except setDecimals() to 0)

Reply:
Just use cm for metric paper sizes, it is more common than mm.

Reply:
Ok. Fixed.